### PR TITLE
Makefile: allows the main file to have .k extension

### PR DESCRIPTION
**Pull request by daejunpark**
This allows e.g., `make build-haskell --haskell_main_file=/path/to/verification.k` that is needed for using kore-repl.

### CODE REVIEW

**Comment by daejunpark** [comment]
If this makes sense, I can change the other build-* targets if needed.

**Comment by daejunpark** [comment]
The second commit has the kevm script to launch kore-repl with ledit, if available, that enables line editing for the repl command line.

File: kevm [hunk 1]
```diff
@@ -52,13 +52,15 @@ run_kast() {
...
+    local def_module run_dir additional_proof_args ledit
 
     def_module="$1" ; shift
 
+    command -v ledit >/dev/null 2>&1 && ledit=ledit
+
     additional_proof_args=()
     ! $debug || additional_proof_args+=(--debug)
-    ! $repl  || additional_proof_args+=(--haskell-backend-command "$k_release_dir/bin/kore-repl --repl-script $kevm_dir/kast.kscript")
+    ! $repl  || additional_proof_args+=(--haskell-backend-command "$ledit $k_release_dir/bin/kore-repl --repl-script $kevm_dir/kast.kscript")
```
Comment (ehildenb) [inline kevm:63]: What is the purpose here?

File: Makefile [hunk 2]
```diff
@@ -213,13 +213,14 @@ haskell_dir            := $(DEFN_DIR)/haskell
...
-haskell_kompiled       := $(haskell_dir)/$(haskell_main_file)-kompiled/definition.kore
+haskell_main_file      := driver.md
+haskell_main_filename  := $(basename $(notdir $(haskell_main_file)))
+haskell_kompiled       := $(haskell_dir)/$(haskell_main_filename)-kompiled/definition.kore
 
 build-haskell: $(haskell_kompiled)
 
 $(haskell_kompiled): $(haskell_files)
-	$(KOMPILE_HASKELL) $(haskell_main_file).md                  \
+	$(KOMPILE_HASKELL) $(haskell_main_file)                     \
```
Comment (ehildenb) [inline Makefile:223]: Yes, let's be consistent across all the definitions.

File: kevm (hunk 1 above)
Comment (daejunpark) [reply kevm]: I made this more general in the new commit, but the motivation is to use `ledit` to enable line editing in the repl command line (like editing commands, search previous commands, etc.).

File: Makefile (hunk 2 above)
Comment (daejunpark) [reply Makefile:225]: done

File: Makefile [hunk 3]
```diff
@@ -230,9 +233,10 @@ web3_dir           := $(abspath $(DEFN_DIR)/web3)
 web3_files         := $(ALL_FILES)
 web3_main_module   := WEB3
 web3_syntax_module := $(web3_main_module)
-web3_main_file     := web3
+web3_main_file     := web3.md
+web3_main_filename := $(basename $(notdir $(web3_main_file)))
 web3_kompiled      := $(web3_dir)/build/kevm-client
-web3_kore          := $(web3_dir)/$(web3_main_file)-kompiled/definition.kore
+web3_kore          := $(web3_dir)/$(web3_main_filename)-kompiled/definition.kore
 export web3_main_file
```
Comment (daejunpark) [inline Makefile:240]: This build seems to fail due to this.  @ehildenb Do you know what I should do for this export?

Comment (daejunpark) [reply Makefile]: nvm

File: kevm [hunk 4]
```diff
@@ -56,9 +56,12 @@ run_prove() {
 
     def_module="$1" ; shift
 
+    repl_cmd="${repl_cmd:-$k_release_dir/bin/kore-repl}"
+    repl_script="${repl_script:-$kevm_dir/kast.kscript}"
```
Comment (ehildenb) [inline kevm:60]:
```suggestion
    repl_cmd="${REPL_CMD:-$k_release_dir/bin/kore-repl}"
    repl_script="${REPL_SCRIPT:-$kevm_dir/kast.kscript}"
```

Also add `repl_cmd` and `repl_script` to the `local ...` declaration above.

Comment (daejunpark) [reply kevm]: sounds good. done.

**Pull request merged**